### PR TITLE
tests: Small cleanup

### PR DESCRIPTION
**Pull request by JadenSimon**
## Problem
There were a few minor issues with the tests, notably a bunch of unhandled rejected promises.  

## Solution
* Remove `assertUtils` and use `assert.rejects`/`assert.throws` instead (this fixed a couple of the unhandled promises) 
* Properly dispose of the template/CodeLens registry in global test setup (this fixed the rest)
* Add 'fake-timers' types to `tsconfig.json` so types actually show up for FakeTimers
* Launch unit tests with extensions disabled (nice little speed-up for locally run tests)

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/shared/sam/cli/samCliInvokerUtils.ts [hunk 1]
```diff
@@ -32,6 +32,10 @@ export interface SamCliProcessInvoker {
     stop(): void
 }
 
+export function makeUnpectedExitCodeError(message: string): Error {
```
Comment (justinmk3) [inline src/shared/sam/cli/samCliInvokerUtils.ts:35]:
```suggestion
export function makeUnexpectedExitCodeError(message: string): Error {
```

File: src/shared/sam/cli/samCliInvokerUtils.ts [hunk 2]
```diff
@@ -56,5 +60,5 @@ export function logAndThrowIfUnexpectedExitCode(processResult: ChildProcessResul
         message = processResult.stderr || processResult.stdout || 'No message available'
     }
 
-    throw new Error(`Error with child process: ${message}`)
+    throw makeUnpectedExitCodeError(message)
```
Comment (justinmk3) [inline src/shared/sam/cli/samCliInvokerUtils.ts:63]:
```suggestion
    throw makeUnexpectedExitCodeError(message)
```

File: .vscode/launch.json [hunk 3]
```diff
@@ -49,6 +49,7 @@
             "request": "launch",
             "runtimeExecutable": "${execPath}",
             "args": [
+                "--disable-extensions",
```
Comment (bryceitoc9) [inline .vscode/launch.json:52]: Also add to "Extension Tests (current file)" below?

File: src/test/lambda/models/samTemplates.test.ts [hunk 4]
```diff
@@ -72,9 +71,11 @@ describe('getSamCliTemplateParameter', function () {
     })
 
     it('should return error if the template option is not valid', async function () {
-        const errorMessage = `${repromptUserForTemplate} is not valid sam template`
-        const error = await assertThrowsError(async () => getSamCliTemplateParameter(repromptUserForTemplate))
-        assert.strictEqual(error.message, errorMessage, 'Should fail for same error')
+        await assert.rejects(
```
Comment (bryceitoc9) [inline src/test/lambda/models/samTemplates.test.ts:74]: Should this be `assert.throws`? `getSamCliTemplateParameter` is not async.

File: src/ssmDocument/commands/openDocumentItem.ts [hunk 5]
```diff
@@ -38,7 +38,9 @@ export async function openDocumentItem(node: DocumentItemNode, awsContext: AwsCo
             content: rawContent.Content,
             language: `ssm-${rawContent.DocumentFormat!.toLowerCase()}`,
         })
-        vscode.window.showTextDocument(textDocument)
+        if (textDocument !== undefined) {
```
Comment (bryceitoc9) [inline src/ssmDocument/commands/openDocumentItem.ts:41]: I think this if statement renders the catch block mostly useless? Which means we won't get a failure event.

File: src/test/lambda/models/samTemplates.test.ts (hunk 4 above)
Comment (JadenSimon) [reply src/test/lambda/models/samTemplates.test.ts:74]: Yup must've missed it.

File: src/ssmDocument/commands/openDocumentItem.ts (hunk 5 above)
Comment (JadenSimon) [reply src/ssmDocument/commands/openDocumentItem.ts:41]: Should be changed to `await` instead of an if statement.

**Review by justinmk3 — APPROVED** [review]
Wonderful

**Pull request merged**